Source/UTF8LineParser.swift

```diff
@@ -31,25 +21,22 @@ class UTF8LineParser {
 
     func append(_ body: Data) -> [String] {
         let data = remainder + body
-        var dataIter = DataIter(data)
+        var dataIter = DataIter(data: data)
         var remainderPos = data.endIndex
         var lines: [String] = []
 
         Decode: while true {
             switch utf8Parser.parseScalar(from: &dataIter) {
             case .valid(let scalarResult):
                 let scalar = Unicode.UTF8.decode(scalarResult)
-                if seenCr {
-                    lines.append(currentString)
-                    currentString = ""
+
+                if seenCr && scalar == lf {
                     seenCr = false
-                    if scalar == lf {
-                        continue
-                    }
+                    continue
                 }
-                if scalar == cr {
-                    seenCr = true
```

Comment:
Another simplification here is that we flush the line immediately when we receive a carriage return. This shouldn't be a spec change, it just ended up being simpler to ignore the following line feed if present, rather than waiting to see whether the next character is a line feed to dispatch the already complete line (since it is complete regardless of whether a line feed follows the carriage return.